packages/bonito-ui/src/components/form/string-list.tsx

```diff
@@ -0,0 +1,170 @@
+        items.push("");
+        return items;
+    }, [param.value]);
+
+    const onItemChange = useCallback(
+        (index: number, value: string) => {
+            if (index === items.length - 1) {
+                // Last item, add a new one
+                items.push("");
+            }
+            items[index] = value;
+            param.value = items.slice(0, items.length - 1) as V[K];
+            onChange?.(null as any, param.value);
+        },
+        [items, param, onChange]
+    );
+
+    const onItemDelete = useCallback(
+        (index: number) => {
+            items.splice(index, 1);
+            param.value = items.slice(0, items.length - 1) as V[K];
+            onChange?.(null as any, param.value);
+        },
+        [items, param, onChange]
+    );
+
+    return (
+        <Stack key={id} style={style} className={className}>
+            {items.map((item, index) => {
+                const errorMsg = validationData?.[index];
+                return (
+                    <StringListItem
+                        key={index}
+                        index={index}
+                        value={item}
+                        errorMsg={errorMsg}
+                        placeholder={param.placeholder}
+                        onChange={onItemChange}
+                        onDelete={onItemDelete}
+                        disableDelete={index === items.length - 1}
+                    ></StringListItem>
+                );
+            })}
+        </Stack>
+    );
+}
+
+interface StringListItemProps {
+    index: number;
+    value: string;
+    onChange: (index: number, value: string) => void;
+    onDelete: (index: number) => void;
+    placeholder?: string;
+    disableDelete?: boolean;
+    errorMsg?: string;
+}
+
+function StringListItem(props: StringListItemProps) {
+    const {
+        index,
+        value,
+        onChange,
+        onDelete,
+        disableDelete,
+        errorMsg,
+        placeholder,
+    } = props;
+    const styles = useStringListItemStyles(props);
+    return (
+        <Stack
+            key={index}
+            horizontal
+            verticalAlign="center"
+            styles={styles.container}
+        >
+            <Stack.Item grow={1}>
+                <TextField
+                    styles={styles.input}
+                    value={value}
+                    placeholder={placeholder}
+                    errorMessage={errorMsg}
+                    onChange={(_, newValue) => {
+                        onChange(index, newValue || "");
+                    }}
+                />
+            </Stack.Item>
+            <IconButton
+                styles={styles.button}
+                iconProps={{ iconName: "Delete" }}
+                onClick={() => {
+                    onDelete(index);
+                }}
+                disabled={disableDelete}
+            />
+        </Stack>
+    );
+}
+
+function useStringListItemStyles(props: StringListItemProps) {
```

Comment:
Should we switch this to use mergeStyles?

Reply:
`mergeStyles` create class names, which needs to add more wrapper `div` around FluentUI components and then apply to it. I found it more convenient to just apply styles to the FluentUI components.

Reply:
Btw, if we want to modify the FluentUI components' internal element's style, we need to pass style object to it